crates/compiler/test_syntax/tests/test_fmt.rs

```diff
@@ -6223,13 +6223,88 @@ mod test_fmt {
                     [first as last]
                     | [first, last] ->
                         first
-                
+
                     _ -> Not
                 "
             ),
         );
     }
 
+    #[test]
+    fn preserve_annotated_body() {
+        expr_formats_same(indoc!(
+            r"
+                x : i32
+                x = 1
+                x
+                "
+        ));
+    }
+
+    #[test]
+    fn preserve_annotated_body_comment() {
+        expr_formats_same(indoc!(
+            r"
+                x : i32 # comment
+                x = 1
+                x
+                "
+        ));
+    }
+
+    #[test]
+    fn preserve_annotated_body_comments() {
+        expr_formats_same(indoc!(
+            r"
+                x : i32
+                # comment
+                # comment 2
+                x = 1
+                x
+                "
+        ));
+    }
+
+    #[test]
+    fn preserve_annotated_body_comments_with_newlines() {
+        expr_formats_same(indoc!(
+            r"
+                x : i32
+
+                # comment
+
+                # comment 2
+
+                x = 1
+                x
+                "
+        ));
+    }
+
+    #[test]
+    fn preserve_annotated_body_blank_comment() {
+        expr_formats_same(indoc!(
+            r"
+                x : i32
+                #
+                x = 1
+                x
+                "
+        ));
+    }
+
+    #[test]
+    fn preserve_annotated_body_with_newlines() {
+        expr_formats_same(indoc!(
+            r"
+                x : i32
+
+                x = 1
+                x
+                "
+        ));
+    }
+
```

Comment:
I'm happy with whatever you all think is best stylistically!  It would be a pretty simple change to not preserve extra newlines:
```diff
diff --git a/crates/compiler/fmt/src/def.rs b/crates/compiler/fmt/src/def.rs
index cdc0837b8..9bbb0b27e 100644
--- a/crates/compiler/fmt/src/def.rs
+++ b/crates/compiler/fmt/src/def.rs
@@ -604,16 +604,17 @@ pub fn fmt_annotated_body_comment<'a>(
         }

         for comment_or_newline in comment_iter {
-            buf.newline();
             match comment_or_newline {
                 roc_parse::ast::CommentOrNewline::Newline => (),
                 roc_parse::ast::CommentOrNewline::DocComment(comment_str) => {
+                    buf.newline();
                     buf.indent(indent);
                     buf.push_str("# #");
                     buf.spaces(1);
                     buf.push_str(comment_str.trim());
                 }
                 roc_parse::ast::CommentOrNewline::LineComment(comment_str) => {
+                    buf.newline();
                     buf.indent(indent);
                     buf.push_str("#");
                     buf.spaces(1);
```

Reply:
> @Anton-4 maybe you have some insights?

I would get rid of empty lines between the type and the definition. We can easily change it later if someone makes a compelling case for keeping them.

Reply:
> I would get rid of empty lines between the type and the definition.

Sounds good!  Changed in 2529fa07215c44052c99bac8912afd3cc4c0bd92